server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java

```diff
@@ -94,12 +100,52 @@ public QProfileDto getProfile(DbSession dbSession, QProfileReference ref) {
     return profile;
   }
 
-  public void checkPermission(DbSession dbSession, QProfileDto rulesProfile) {
-    OrganizationDto organization = getOrganization(dbSession, rulesProfile);
+  public QProfileDto getProfile(DbSession dbSession, OrganizationDto organization, String name, String language) {
+    QProfileDto profile = dbClient.qualityProfileDao().selectByNameAndLanguage(dbSession, organization, name, language);
+    checkFound(profile, "Quality Profile for language '%s' and name '%s' does not exist in organization '%s'", language, name, organization.getKey());
+    return profile;
+  }
+
+  public UserDto getUser(DbSession dbSession, OrganizationDto organization, String login) {
+    UserDto user = dbClient.userDao().selectActiveUserByLogin(dbSession, login);
+    checkFound(user, "User with login '%s' is not found'", login);
+    checkMembership(dbSession, organization, user);
+    return user;
+  }
+
+  public GroupDto getGroup(DbSession dbSession, OrganizationDto organization, String groupName) {
+    Optional<GroupDto> group = dbClient.groupDao().selectByName(dbSession, organization.getUuid(), groupName);
+    checkFoundWithOptional(group, "No group with name '%s' in organization '%s'", groupName, organization.getKey());
+    return group.get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "group.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)